doc/users/style_sheets.rst

```diff
@@ -25,15 +25,15 @@ Defining your own style
 =======================
 
 You can create custom styles and use them by calling ``style.use`` with the
-path or URL to the style sheet. Alternatively, if you add your
-``<style-name>.mplstyle`` file to ``~/.matplotlib/stylelib`` (you may need to
-create this directory), you can reuse your custom style sheet with a call to
-``style.use(<style-name>)``. Note that a custom style sheet in
-``~/.matplotlib/stylelib`` will override a style sheet defined by matplotlib if
-the styles have the same name.
+path or URL to the style sheet. Alternatively, if you add your ``<style-name>.mplstyle`` 
+file to ``mpl_configdir/stylelib`, you can reuse your custom style sheet with a call to 
+``style.use(<style-name>)``. By default ``mpl_configdir`` should be ``~/.config/matplotlib``, 
```

Comment:
On my system (OSX), this is actually `~/.matplotlib`. There's probably no general way to explicitly write the path for all system. Maybe just writing `<mpl_configdir>/stylelib` and telling users how to find `<mpl_configdir>` would be sufficient.